wgsl/index.bs

```diff
@@ -3538,7 +4028,61 @@ compound_statement
   : BRACE_LEFT statements BRACE_RIGHT
 </pre>
 
-## Assignment TODO ## {#assignment}
+## Assignment ## {#assignment}
+
+An <dfn noexport>assignment statement</dfn> replaces the contents of a variable,
+or a portion of a variable, with a new value.
+
+The
+expression to the left of the equals token is the <dfn noexport>left-hand side</dfn>,
+and the
+expression to the right of the equals token is the <dfn noexport>right-hand side</dfn>.
+
+<table class='data'>
+  <thead>
+    <tr><th>Precondition<th>Statement<th>Description
+  </thead>
+  <tr algorithm="assignment">
+    <td>|r| : ref<|SC|,|T|>,<br>
+        |e| : |T|,<br>
+        |T| is [=storable=],<br>
```

Comment:
and writable?

Reply:
ah.  this would be "does not have access(read)"

But if it's writing to a part of a storage buffer, it's really talking about the store type of the whole buffer, and that type may include |T|.
Basically, this shows that having access attribute on type is broken.  It belongs on the originating variable.
I'll add a TODO for this. Thanks!

Reply:
There's a validation statement immediately after this table:

> The [=originating variable=] of the left-hand side must not have an `access(read)` access attribute.  

But it's really the *store* type of the originating variable....